app/src/main/java/org/dhis2/uicomponents/map/geometry/line/MapLineRelationshipToFeature.kt

```diff
@@ -27,4 +32,85 @@ class MapLineRelationshipToFeature {
         }
         return null
     }
+
+    private fun getPoint(geo: Geometry): List<Double> {
+        return when (geo.type()) {
+            FeatureType.POINT -> GeometryHelper.getPoint(geo)
+            FeatureType.POLYGON -> GeometryHelper.getPolygon(geo)[0][0]
+            else -> {
+                arrayListOf(0.0, 0.0)
+            }
+        }
+    }
+
+    private fun getStartPoints(
+        fromGeometry: Geometry,
+        toGeometry: Geometry
+    ): Pair<List<Double>, List<Double>> {
+        return if (fromGeometry.type() == FeatureType.POINT &&
+            toGeometry.type() == FeatureType.POINT
+        ) {
+            Pair(GeometryHelper.getPoint(fromGeometry), GeometryHelper.getPoint(toGeometry))
+        } else if (fromGeometry.type() == FeatureType.POINT &&
+            toGeometry.type() == FeatureType.POLYGON
+        ) {
+            Pair(
+                GeometryHelper.getPoint(fromGeometry),
+                getPolygonCloserPoint(
```

Comment:
I miss the unit test of this part. To make it simpler I would create another class to compute this part

Reply:
done